lib/features/thread/presentation/widgets/scroll_to_top_button_widget.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:core/presentation/utils/responsive_utils.dart';
+import 'package:flutter/material.dart';
+import 'package:tmail_ui_user/features/thread/presentation/styles/scroll_to_top_button_widget_styles.dart';
+
+class ScrollToTopButtonWidget extends StatefulWidget {
+  final ScrollController scrollController;
+  final GestureTapCallback onTap;
+  final ResponsiveUtils responsiveUtils;
+  final double? limitIndicator;
+  final double? elevation;
+  final Color? colorButton;
+  final double? buttonRadius;
+  final Widget? icon;
+
+  const ScrollToTopButtonWidget({
+    Key? key,
+    required this.scrollController,
+    required this.onTap,
+    required this.responsiveUtils,
+    this.limitIndicator = ScrollToTopButtonWidgetStyles.defaultLimitIndicator,
+    this.elevation = ScrollToTopButtonWidgetStyles.defaultElevation,
+    this.colorButton = ScrollToTopButtonWidgetStyles.defaultColor,
+    this.buttonRadius = ScrollToTopButtonWidgetStyles.defaultButtonRadius,
+    this.icon,
+  }) : super(key: key);
+
+  @override
+  State<ScrollToTopButtonWidget> createState() => _ScrollToTopButtonWidgetState();
+}
+
+class _ScrollToTopButtonWidgetState extends State<ScrollToTopButtonWidget> {
+  bool _isVisible = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _handleScroll();
+  }
+
+  @override
+  void dispose() {
+    widget.scrollController.removeListener(() {});
```

Comment:
Should be delete the `specific listener` function to avoid the situation where we have many listeners here later